src/main/java/uk/gov/companieshouse/confirmationstatementapi/controller/ConfirmationStatementController.java

```diff
@@ -36,41 +40,76 @@ public ConfirmationStatementController(ConfirmationStatementService confirmation
     }
 
     @PostMapping("/")
-    public ResponseEntity<Object> createNewSubmission(@RequestAttribute("transaction") Transaction transaction, HttpServletRequest request) {
+    public ResponseEntity<Object> createNewSubmission(
+            @RequestAttribute("transaction") Transaction transaction,
+            @PathVariable(TRANSACTION_ID_KEY) String transactionId,
+            HttpServletRequest request) {
+
+        String passthroughHeader = request.getHeader(ApiSdkManager.getEricPassthroughTokenHeader());
+        String requestId = request.getHeader(ERIC_REQUEST_ID_KEY);
+
+        var logMap = new HashMap<String, Object>();
+        logMap.put(TRANSACTION_ID_KEY, transactionId);
+        LOGGER.infoContext(requestId, "Calling service to create submission", logMap);
 
-        String passthroughHeader = request
-                .getHeader(ApiSdkManager.getEricPassthroughTokenHeader());
         try {
             return confirmationStatementService.createConfirmationStatement(transaction, passthroughHeader);
         } catch (ServiceException e) {
-            LOGGER.error("Error Creating Confirmation Statement", e);
+            LOGGER.errorContext(requestId,"Error Creating Confirmation Statement", e, logMap);
             return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
         }
     }
 
     @PostMapping("/{confirmation_statement_id}")
-    public ResponseEntity<Object> updateSubmission(@RequestBody ConfirmationStatementSubmissionJson confirmationStatementSubmissionJson,
-                                                   @PathVariable("confirmation_statement_id") String submissionId) {
+    public ResponseEntity<Object> updateSubmission(
+            @RequestBody ConfirmationStatementSubmissionJson confirmationStatementSubmissionJson,
+            @PathVariable(CONFIRMATION_STATEMENT_ID_KEY) String submissionId,
+            @PathVariable(TRANSACTION_ID_KEY) String transactionId,
+            @RequestHeader(value = ERIC_REQUEST_ID_KEY) String requestId) {
+
+        var logMap = new HashMap<String, Object>();
+        logMap.put(CONFIRMATION_STATEMENT_ID_KEY, submissionId);
+        logMap.put(TRANSACTION_ID_KEY, transactionId);
+        LOGGER.infoContext(requestId, "Calling service to update confirmation statement data", logMap);
```

Comment:
Don't think there is and this is a pattern across all controllers. If needed, can we create a separate ticket for it? Otherwise this one will go out of scope and the PR will get even bigger.

Reply:
Sounds like we will need to add some confirmation logging but I agree probably best in a separate PR. Could do in a separate PR but under same ticket.

Reply:
Creating [NCS-776](https://companieshouse.atlassian.net/browse/NCS-776) address the logging review on both the [URL](https://github.com/companieshouse/confirmation-statement-api/pull/160#discussion_r728945982) and [confirmation](https://github.com/companieshouse/confirmation-statement-api/pull/160#discussion_r728985545) logging suggestions.